registry/storage/garbagecollect.go

```diff
@@ -15,14 +15,16 @@ func emit(format string, a ...interface{}) {
 }
 
 // MarkAndSweep performs a mark and sweep of registry data
-func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, registry distribution.Namespace, dryRun bool) error {
+func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, registry distribution.Namespace, dryRun bool, manifestswithouttags bool) error {
 	repositoryEnumerator, ok := registry.(distribution.RepositoryEnumerator)
 	if !ok {
 		return fmt.Errorf("unable to convert Namespace to RepositoryEnumerator")
 	}
 
 	// mark
 	markSet := make(map[digest.Digest]struct{})
+	manifestSet := make(map[digest.Digest]string)
```

Comment:
`manifestDeleteSet`

Make sure to have a comment here that we are storing in the target repository.